cmd/vodloadtester/vodloadtester.go

```diff
@@ -0,0 +1,423 @@
+					vt.writeResultNdjson(uploadTest)
+					return
+				} else {
+					uploadTest.RequestUploadSuccess = 1
+					uploadTest.AssetID = requestedUpload.Asset.ID
+					uploadTest.TaskID = requestedUpload.Task.ID
+				}
+
+				err = vt.uploadAsset(fileName, requestedUpload.Url)
+
+				if err != nil {
+					glog.Errorf("Error uploading asset: %v", err)
+					uploadTest.UploadSuccess = 0
+				} else {
+					uploadTest.UploadSuccess = 1
+					if vt.cliFlags.TaskCheck {
+						err := vt.checkTaskProcessing(5*time.Second, api.TaskOnlyId{ID: uploadTest.TaskID})
+						if err != nil {
+							uploadTest.TaskCheckSuccess = 0
+							uploadTest.ErrorMessage = err.Error()
+						} else {
+							uploadTest.TaskCheckSuccess = 1
+						}
+					}
+
+				}
+				vt.writeResultNdjson(uploadTest)
+				wg.Done()
+			}()
+		}
+		time.Sleep(vt.cliFlags.StartDelayDuration)
+	}
+
+	wg.Wait()
+}
+
+func (vt *vodLoadTester) resumableUploadLoadTest(fileName, runnerInfo string) {
+	wg := &sync.WaitGroup{}
+
+	for i := 0; i < int(vt.cliFlags.VideoAmount); i += int(vt.cliFlags.Simultaneous) {
+		for j := 0; j < int(vt.cliFlags.Simultaneous); j++ {
+			fmt.Printf("rUpload video %d/%d\n", i+j+1, vt.cliFlags.VideoAmount)
+			wg.Add(1)
+			go func() {
+				uploadTest := uploadTest{
+					StartTime:  time.Now(),
+					RunnerInfo: runnerInfo,
+					Kind:       "resumable",
+				}
+				rndAssetName := fmt.Sprintf("load_test_resumable_%s", randName())
+				requestedUpload, err := vt.requestUploadUrls(rndAssetName)
+				defer vt.lapi.DeleteAsset(requestedUpload.Asset.ID)
+
+				if err != nil {
+					glog.Errorf("Error requesting upload urls: %v", err)
+					uploadTest.RequestUploadSuccess = 0
+					uploadTest.ErrorMessage = err.Error()
+					vt.writeResultNdjson(uploadTest)
+					return
+				} else {
+					uploadTest.RequestUploadSuccess = 1
+					uploadTest.AssetID = requestedUpload.Asset.ID
+					uploadTest.TaskID = requestedUpload.Task.ID
+				}
+
+				uploadUrl := requestedUpload.TusEndpoint
+
+				err = vt.uploadAssetResumable(uploadUrl, fileName)
+
+				if err != nil {
+					glog.Errorf("Error on resumable upload: %v", err)
+					uploadTest.UploadSuccess = 0
+				} else {
+					uploadTest.UploadSuccess = 1
+					if vt.cliFlags.TaskCheck {
+						err := vt.checkTaskProcessing(5*time.Second, api.TaskOnlyId{ID: uploadTest.TaskID})
+						if err != nil {
+							uploadTest.TaskCheckSuccess = 0
+							uploadTest.ErrorMessage = err.Error()
+						} else {
+							uploadTest.TaskCheckSuccess = 1
+						}
+					}
+
+				}
+				vt.writeResultNdjson(uploadTest)
+				wg.Done()
+			}()
+		}
+		time.Sleep(vt.cliFlags.StartDelayDuration)
+	}
+
+	wg.Wait()
+
+}
+
+func (vt *vodLoadTester) importFromUrlTest(url string, runnerInfo string) {
+	wg := &sync.WaitGroup{}
+
+	for i := 0; i < int(vt.cliFlags.VideoAmount); i += int(vt.cliFlags.Simultaneous) {
+		for j := 0; j < int(vt.cliFlags.Simultaneous); j++ {
+			fmt.Printf("Importing video %d/%d\n", i+j+1, vt.cliFlags.VideoAmount)
+			wg.Add(1)
```

Comment:
This has the same problems stated on the above function. Gotta fix the concurrency and the usage of `sync.WaitGroup`.